### DISCUSSION TITLE
How to handle templater scripts eating selection

### DISCUSSION THREAD

**Discussion by scwunch** [SilentVoid13/Templater#419] [Ideas & Feature Requests]
I was writing a templater script to modify text — but only selected text — in the active file.  I noticed that, no matter what I did, it just eats up my selection and doesn't give it back.  At first I was confused, but then I realized that Templater actually *always* eats my selection — no matter what function is called — and doesn't give it back.  I could manually bring it back with a `tR += previouslySelectedText` or `editor.replaceSelection(previouslySelectedText)` or something like that, but then my cursor is back to caret mode rather than selection mode.

I ended up "solving" this problem with this snippet of code: 
```
const editor = app.workspace.activeLeaf.view.editor;
let selection = editor.getSelection();
let cAnchor = editor.getCursor('anchor');
let cHead = editor.getCursor('head');
setTimeout(() => editor.setSelection(cAnchor, cHead), 0);
editor.replaceSelection(selection);
```

I don't know why `setTimeout(... 0)` works, but it does.  I'm curious also if there are other better workarounds.

**Comment by FelipeRearden** [comment]
Hello @scwunch !!!

My deepest thank you for sharing this Snippet. I was struggling with a workflow related to text selection for almost a year and your snippet save my Obsidian life. 💙💙💙

**Reply by scwunch** [reply]
Why thank you for the appreciations @FelipeRearden ; I'm glad that this snippet was able to make your life easier :smile:
